### ISSUE TITLE
Looking for feedback on game logic in reducer

### ISSUE THREAD

**Issue by ZackWard** [ZackWard/react-tic-tac-toe#1]
I hope your afternoon is going well Alex,

I was just thinking about the game logic, and it seems like the reducer is not the ideal place for it since it includes some randomness. I'm thinking about moving the game logic into an action creator. Any thoughts?

**Comment by ZackWard** [comment]
I went ahead and moved the logic into the action creator. Hopefully that will keep the reducer clean and be more in line with standard practices. (no side effects in the reducer)

**Comment by zealoushacker** [comment]
I think it depends on the logic. Can you point me to what part of the game logic you are thinking about moving/have already moved?

**Comment by ZackWard** [comment]
Here is what I settled on at the end of the day today: 

https://github.com/ZackWard/react-tic-tac-toe/blob/a1b70b2b059e853b73ead58a34a501a8e20b0dbb/src/actionCreators.js#L37-L106

However, after thinking about it a little further, I think that I'm overcomplicating things. I was trying to make each move a discreet state change, so that when the player makes a move, it would involve dispatching several actions. (dispatch player move, calculate computer move, dispatch that, etc) My thinking was that by doing so, I could undo each move individually. Undo the player move, then undo the computer move, etc. 

Instead, I'm considering wrapping the state changes for the player to move, and then for the computer to move, into one action. I realized that there isn't a lot of utility in undoing the computer's move only to re-do it. If the player undos the last move, they will probably want to undo the computer's move, and their move. That way, they can choose a different move.
